src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -865,13 +867,43 @@ internal static string ResolvePackagePath(string searchDirectory, string searchP
             {
                 // If the search does not contain wild cards, and the target path shares the same extension, copy it
                 // e.g. <file src="ie\css\style.css" target="Content\css\ie.css" /> --> Content\css\ie.css
-                return targetPath;
+                packagePath = targetPath;
+                targetPath = null;
             }
             else
             {
                 packagePath = Path.GetFileName(fullPath);
             }
-            return Path.Combine(targetPath ?? String.Empty, packagePath);
+
+            string path = Path.Combine(targetPath ?? string.Empty, packagePath);
+
+            // Translate the tfm alias to its actual framework if necessary.
+            foreach (string knownFolder in PackagingConstants.Folders.Known)
+            {
+                string folderPrefix = knownFolder + Path.DirectorySeparatorChar;
+                if (path.Length > folderPrefix.Length &&
+                    path.StartsWith(folderPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    string frameworkPart = path.Substring(folderPrefix.Length);
+                    string targetFrameworkString = Path.GetDirectoryName(frameworkPart).Split(Path.DirectorySeparatorChar).First();
```

Comment:
pack performance is no where near as critical as restore performance. But, it would be much higher performance to find the first index of `Plath.DirectorySeparatorChar` whose index is greater than `folderPrefix.Length`, and then substring that segment.